examples/with-auth/components/session.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * A class to handle signing in and out and caching session data in sessionStore
+ *
+ * Note: We usewindow.XMLHttpRequest() here rather than fetch because fetch() uses
+ * Service Workers and they cannot share cookies with the browser session
+ * yet (!) so if we tried to get or pass the CSRF token it would mismatch.
```

Comment:
are you talking about "credentials" option of fetch? it allows send cookies also. or i misunderstood smthing...
```
fetch(url, {  
  credentials: 'include'  
})
```